import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ImportAdapter.java

```diff
@@ -121,10 +121,16 @@ public static void main(String[] args) throws IOException, GeneralSecurityExcept
     BackupUploadService backupUploadService = configureBackupUploadService(flags, credentials);
 
     IDestinationClientFactory destinationClientFactory = configureDestinationClientFactory(
-        defaultCstoreDicomWebClient, credentials, flags, backupUploadService != null);
+      defaultCstoreDicomWebClient, credentials, flags, backupUploadService != null);
 
-    MultipleDestinationUploadService multipleDestinationSendService = configureMultipleDestinationUploadService(
-        flags, cstoreSubAet, backupUploadService);
+    // NOTE: This is a dependency for cStoreService if stow retry is used or if multiple
+    // destinations are leveraged, due to the backupUploadService dependency.
+    MultipleDestinationUploadService multipleDestinationSendService = null;
+    if (flags.sendToAllMatchingDestinations) {
```

Comment:
Require explicitly setting this flag for `useStowOverwrite` to work properly, as it relied on the backupUploadService.

Reply:
Trying to reason about this change a bit: it looks like the state before was, we set the multipleDestinationSendService to null only if the backupService was null. Now we set it to null if sendToAllMatchingDestinations is false. 

I'm wondering, isn't this a breaking change in some ways? E.g. for current users who set up the backupService but do not configure sendToAllMatchingDestinations, won't their backups break now?

Reply:
Reverted the initialization logic and added a check for multiple destinations + stow retry later in the code.